src/test/java/by/matsukiryna/arraytask/service/impl/ArrayServiceImplTest.java

```diff
@@ -0,0 +1,95 @@
+package by.matsukiryna.arraytask.service.impl;
+
+import by.matsukiryna.arraytask.entity.ArrayEntity;
+import by.matsukiryna.arraytask.exception.ArrayException;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertThrows;
+
+public class ArrayServiceImplTest {
+
+    private final ArrayServiceImpl service = new ArrayServiceImpl();
+
+   /* @DataProvider(name = "findAverage")
+    public Object[][] findAverage(){
+        return new Object[][] {
+                //{arrayEntity.getArrayEntity(), true}
+        };
+    }*/
+
+    public ArrayServiceImplTest() throws ArrayException {
+    }
+
+    @Test
+    public void testReplace() throws ArrayException {
+        final ArrayEntity arrayEntity = new ArrayEntity(new int[] {1, 0, -4, 3, -3});
+        int[] actual = service.replace(arrayEntity, 0, 99);
```

Comment:
you should never use actual method implementations in unit tests as mistakes in such cases will never be found (or they can occur because of actual method  calls. Unit tests must  simulate expected results in stead of actual method calls